configure-env.js

```diff
@@ -0,0 +1,97 @@
+const fs = require('fs');
+const path_dev = './packages/backend/.env.development';
+const path_prod = './packages/backend/.env.production';
+
+/**
+ * configure the environment for .env.development and .env.production if not existing
+ * @param  {} filePath - The path of environment files in backend
+ */
+const configureEnvironmentFile = (filePath) => {
+  console.log(`Checking the existence of ${filePath}`);
+
+  fs.access(filePath, fs.F_OK, (err) => {
+    // If this is not case because of not found files, perform early return
+
+    if (err) {
+      console.error(
+        `An unexpected error ${err} is encountered \n'${filePath}' file does not exist, creating from '${filePath}.template '`,
+      );
+
+      // Check the existence of template file
+      fs.access(`${filePath}.template`, fs.F_OK, (err) => {
+        if (err) {
+          console.error(
+            `Fail to find the template file '${filePath}.template' for '${filePath}'.`,
+          );
+        }
+        return;
+      });
+
+      // copy the data from template file and paste to wanted files
+      let data = fs.readFileSync(`${filePath}.template`, 'utf-8', (err) => {
+        if (err) {
+          console.error(`Fail to read data from '${filePath}'.template`);
+        }
+        return;
+      });
+      fs.writeFileSync(filePath, data, (err) => {
+        if (err) {
+          console.error(`Fail to write to '${filePath}' file`);
+        }
+        return;
+      });
+
+      console.log(`'${filePath}' file created successfully`);
+    } else {
+      console.log(`${filePath} exists`);
+      validateEnvironmentFile(filePath);
+    }
+  });
+};
+
+/**
+ * compare the content between environment files and environment template files
+ * @param  {} filePath - The path of environment files in backend
+ */
+const validateEnvironmentFile = (filePath) => {
+  console.log(`Starting to validate ${filePath}`);
+
+  // Checks the existence of the environment template file
+  fs.access(`${filePath}.template`, fs.F_OK, (err) => {
+    if (err) {
+      console.error(
+        `Fail to find the template file '${filePath}.template' for '${filePath}'.`,
+      );
+    }
+    return;
+  });
+
+  // Read the data from the environment file and its template
+  let data = fs.readFileSync(`${filePath}`, 'utf-8', (err) => {
+    if (err) {
+      console.error(`Fail to read data from '${filePath}'`);
+    }
+    return;
+  });
+  let dataTemplate = fs.readFileSync(`${filePath}.template`, 'utf-8', (err) => {
+    if (err) {
+      console.error(`Fail to read data from '${filePath}'.template`);
+    }
+    return;
+  });
+
+  //Compare the existing file to its template
+  if (data.toString() !== dataTemplate.toString()) {
+    console.error(
+      `There is a difference between '${filePath}' and '${filePath}.template', make sure you have the correct configuration in '${filePath}'`,
+    );
+    return;
+  }
```

Comment:
Why check if the utf8 of the two files is the same. The template and the `.env.xxx` files are meant to be different when you configure the template. I think this could be useful if you are missing some environment variables (ie: they are present in the template but not present in the `.env.xxx`) but I think this can check just be skipped. If anything it can just be a error if the files are the same (ie: not configured).

I think there's always going to be some initial setup friction because we have secret keys that can't be put anywhere.

Reply:
Thanks for clarifying👍. I previously thought the template is used to hold all the up to date config, so the script will compare existing scripts and make sure they are the same as the template. In this sense, I will remove all the functionalities in this utf8 checking part.

Reply:
I think you should just get all the keys in the template, and all the keys in the actual env. Then ensure all the keys in the template are present in the actual.